uk.co.compucorp.civicrm.tasksassignments/js/test/filters/unique.filter.spec.js

```diff
@@ -0,0 +1,29 @@
+/* eslint-env amd, jasmine */
+
+define([
+    'common/lodash',
+    'mocks/data/key-dates',
+    'common/angularMocks',
+    'tasks-assignments/modules/tasks-assignments.dashboard.module'
+  ], function (_, keyDatesMock) {
+    'use strict';
+
+    describe('unique', function () {
+      var filter, withDateContactList, withoutDateContactList;
+
+      beforeEach(module('tasks-assignments.dashboard'));
+      beforeEach(inject(function ($filter) {
+        withDateContactList = keyDatesMock.withDateContactList();
+        withoutDateContactList = keyDatesMock.withoutDateContactList();
+
+        filter = $filter('unique');
+      }));
+
+      describe('when filtering array of data by a date', function () {
```

Comment:
This test is not enough. This filter is a pure function, a straightforward algorithm, it needs to be tested similar to this: https://github.com/civicrm/civihr/pull/2238/files#diff-4f7c3ab98c5b2837916a2da49d5b4894R30

```js
// semi-pseudo code

var testCases = [
  // [input, expectedResult, uniqueBy]
  // Common cases
  [['a', 'b', 'c'], ['a', 'b', 'c']],
  [['a', 'd', 'a', 'd', 'c', 'a'], ['a', 'd', 'c']],
  [[1, 2, 3, 3, 1, 2], [1, 2, 3]],
  [[{ id: 1 }, { id: 2 }, { id: 1 }], [{ id: 1 }, { id: 2 }], 'id'],

  // Edge cases
  [[], []],
  ['hello world', 'helo wrd'],
  [{}, ????], // not sure what should happen here tho
  [[undefined, undefined], [undefined]], // not sure tho, may be it should be just []
  [[{ id: 1 }, { id: 2 }, { id: 1 }], [{ id: 1 }, { id: 2 }, { id: 1 }], 'date'],
];

testCases.forEach(function (testCase) {
  describe('...', function () {
    it('transforms ' + JSON.stringify(testCase[0]) + ' to ' + JSON.stringify(testCase[1]), function () {
      expect(filter(testCase[0])).toEqual(testCase[1]);
    });
  });
});
```

Reply:
@bmnepali forgot to mention, I think it would be expected to see more test cases than presented in the snippet above. The snippet is just for the info.